drafts/Stormless3.md

```diff
@@ -0,0 +1,322 @@
+
+The `stepFunctions` block is here to define ... drumroll ... you have already guessed - StepFunctions. Mine is called"`signup`". The `events` section here looks, acts and servers just like `events` in function definition: it configures an API Gateway endpoint for invoking StepFunction from outside of AWS. We'll use it later in a web form. The definition takes. It can be written as YAML right here in `serverless.yml`, 
+but I prefer to include it from a separate file to keep logic separate from configuration. Here it is: 
+
+
+https://gist.github.com/dzimine/94f1a6dc1ba21b79ad09b24711a855b8
+
+<script src="https://gist.github.com/dzimine/
+94f1a6dc1ba21b79ad09b24711a855b8.js"></script>
+
+
+
+```
+# ./stepfunction.yml
+# 
+# StepFunction workflow definition file.
+# Included in `serverless.yml`; variables are defined there.
+
+Comment: "Community sign-up workflow"
+StartAt: RecordDB
+States:
+  RecordDB:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-dev-RecordDB
+    Next: RecordAC
+    ResultPath: $.results.RecordDB
+  RecordAC:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-dev-RecordAC
+    Next: InviteSlack
+    ResultPath: $.results.RecordAC
+  InviteSlack:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-${self:custom.stage}-InviteSlack
+    ResultPath: $.results.InviteSlack
+    End: true
+
+```
+
+
+The StepFunction definition is written in [Amazon States Language](https://states-language.net/spec.html); the spec is short, well written and worth a read. Using YAML instead of JSON is a nice perk from the plugin - it reads better and allows comments. But if you want JSON - no problem, help yourself.
+
+`Resource` is used to refer the Lambda function by ARN. I used the variables we defined above to construct the ARSs matching account ID, region, and stage.
+
+`arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-${self:custom.stage}-RecordDB`
+
+`ResultPath` is used to pass data between steps. By default, StepFunction works on the "need-to-know" basis: the step downstream receives the output from the step directly upstream. If you think it logical, think again: how does RecordAC and InviteSlack are gonna get contact's email? RecordDB may just return "200 OK". Forcing functions re-emit their input would introduce inappropriate intimacy. The trick is to use `ResultPaht` to append the output of each function under some key, like `ResultPath: $results.RecordDB`. Overriding `ResultPath` preserves initial workflow input for use from other Lambda steps. To fully grasp it, read the ResultPath section in the docs, or enjoy a ... < TODO: post from > 
```

Comment:
> how does RecordAC and InviteSlack are gonna get contact's email?

A *little* informal: you may rewrite to "How will RecordAC and InviteSlack get the contact's email address?"

Reply:
Correct `ResultPaht ` to `ResultPath `

Reply:
Don't forget to resolve this TODO